presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionInterpreter.java

```diff
@@ -982,18 +982,25 @@ protected Object visitLambdaExpression(LambdaExpression node, Object context)
             }
 
             Expression body = node.getBody();
-            List<String> argumentNames = node.getArguments().stream()
-                    .map(LambdaArgumentDeclaration::getName)
-                    .collect(toImmutableList());
             FunctionType functionType = (FunctionType) expressionTypes.get(node);
-            checkArgument(argumentNames.size() == functionType.getArgumentTypes().size());
+            List<Class<?>> argumentTypes =
+                    Stream.concat(
+                            Stream.of(ConnectorSession.class),
+                            functionType.getArgumentTypes().stream()
+                                    .map(Type::getJavaType)
+                                    .map(Primitives::wrap))
+                            .collect(toImmutableList());
+            List<String> argumentNames =
+                    Stream.concat(
+                            Stream.of("$connector_session"),
```

Comment:
Looks hacking to hard code this. But not sure if we should put a private static variable in this huge class.